gui/src/app/state/spend/step.rs

```diff
@@ -151,25 +155,68 @@ impl DefineSpend {
         self
     }
 
-    fn check_valid(&mut self) {
-        self.is_valid = self.feerate.valid
+    fn form_values_are_valid(&self) -> bool {
+        self.feerate.valid
             && !self.feerate.value.is_empty()
-            && (self.batch_label.valid || self.recipients.len() < 2);
+            && (self.batch_label.valid || self.recipients.len() < 2)
+            // Recipients will be empty for self-send.
+            && self.recipients.iter().all(|r| r.valid())
+    }
+
+    fn check_valid(&mut self) {
+        self.is_valid =
+            self.form_values_are_valid() && self.coins.iter().any(|(_, selected)| *selected);
         self.is_duplicate = false;
-        if !self.coins.iter().any(|(_, selected)| *selected) {
-            self.is_valid = false;
-        }
         for (i, recipient) in self.recipients.iter().enumerate() {
-            if !recipient.valid() {
-                self.is_valid = false;
-            }
             if !self.is_duplicate && !recipient.address.value.is_empty() {
                 self.is_duplicate = self.recipients[..i]
                     .iter()
                     .any(|r| r.address.value == recipient.address.value);
             }
         }
     }
+    fn auto_select_coins(&mut self, daemon: Arc<dyn Daemon + Sync + Send>) {
+        // Set non-input values in the same way as for user selection.
+        let mut outputs: HashMap<Address<address::NetworkUnchecked>, u64> = HashMap::new();
+        for recipient in &self.recipients {
+            outputs.insert(
+                Address::from_str(&recipient.address.value).expect("Checked before"),
+                recipient.amount().expect("Checked before"),
+            );
+        }
+        let feerate_vb = self.feerate.value.parse::<u64>().unwrap_or(0);
+        // Create a spend with empty inputs in order to use auto-selection.
+        match daemon.create_spend_tx(&[], &outputs, feerate_vb) {
```

Comment:
One issue with creating a spend on every update is how the change address derivation index will keep being increased over and over. I think that's fine for now.

Reply:
After #821, we could call `create_spend` here with a fixed change address just for coin selection purposes, and then increment the change index only when the user clicks "Next" and the PSBT is generated on screen.